ext/opcache/jit/zend_jit_ir.c

```diff
@@ -17025,9 +17165,21 @@ static void jit_frameless_icall2(zend_jit_ctx *jit, const zend_op *opline, uint3
 	if (op2_info & MAY_BE_REF) {
 		op2_ref = jit_ZVAL_DEREF_ref(jit, op2_ref);
 	}
+
+	ir_ref skip_observer = IR_UNUSED;
+	if (ZEND_OBSERVER_ENABLED) {
+		skip_observer = jit_frameless_observer(jit, opline);
+	}
+
 	ir_CALL_3(IR_VOID, ir_CONST_ADDR((size_t)function), res_ref, op1_ref, op2_ref);
+
+	if (skip_observer != IR_UNUSED) {
+		ir_MERGE_WITH(skip_observer);
+	}
+
 	jit_FREE_OP(jit, opline->op1_type, opline->op1, op1_info, NULL);
 	/* Set OP1 to UNDEF in case FREE_OP2() throws. */
+	// TODO: I believe this is only necessary when jit_ir_needs_dtor is true for either op
 	if ((opline->op1_type & (IS_VAR|IS_TMP_VAR)) != 0 && (opline->op2_type & (IS_VAR|IS_TMP_VAR)) != 0) {
 		jit_set_Z_TYPE_INFO(jit, op1_addr, IS_UNDEF);
 	}
```

Comment:
I didn't completely understand your comment about ``jit_ir_needs_dtor``. We don't have such variable or function.
I agree, these IS_UNDEF assignments may be optimized out if we take into account types inferred for next operands and prove that their destruction can't cause exceptions.

@bwoebi please remove this comment

@iluuu1994 please think about the proposed optimization

Reply:
I'll remove it - I don't even recall the exact details when I wrote that.